src/Controller/NphParticipantSummaryController.php

```diff
@@ -36,6 +36,10 @@ public function index(
         $participant = $nphParticipantSummaryService->getParticipantById($participantId);
         $agreeForm = $this->createForm(NphCrossSiteAgreeType::class, null);
         $agreeForm->handleRequest($request);
+        if ($participant === false) {
+            $this->addFlash('error', 'Participant ID not found');
+            return $this->redirectToRoute('nph_participants');
+        }
```

Comment:
@Ehaic Instead of redirecting we could just throw not found exception.
```
        if (!$participant) {
            throw $this->createNotFoundException();
        }
```
Also, could you place this check immediately after getting the participant. i.e after line 36